apps/dashboard_web/lib/dashboard_web/live/components/headcounts.ex

```diff
@@ -9,26 +9,33 @@ defmodule DashboardWeb.Components.Headcounts do
 
   @impl true
   def update(assigns, socket) do
-    headcounts = Dashboard.Stores.get(genserver_id(assigns), "headcounts")
+    id = genserver_id(assigns, assigns.dashboard_component)
+    headcounts = Dashboard.Stores.get(id, "headcounts")
 
-    {:ok, assign(socket, :headcounts, headcounts)}
+    {:ok, socket |> assign(:headcounts, headcounts) |> assign(:genserver_id, id)}
   end
 
   @impl true
   def render(assigns) do
+    response = Dashboard.Stores.get(assigns.genserver_id, :last_response)
+
+    total_count = response["meta"]["total_count"]
+
     assigns =
       assigns
       |> Map.put(:title, "Headcounts")
       |> Map.put(:product, :checkins)
       |> Map.put(:grid_width, 1)
       |> Map.put(:data_key, :headcounts)
-      |> Map.put(:data_number, length(assigns.headcounts))
+      |> Map.put(:data_number, total_count)
 
     DashboardWeb.LayoutView.render("number-card.html", assigns)
   end
 
   @impl DashboardWeb.Behaviours.ComponentLiveView
   def genserver_id(assigns, dc \\ %Dashboard.Dashboards.DashboardComponent{}) do
-    "headcounts--user_#{assigns.user_id}"
+    suffix = Dashboard.Dashboards.DashboardComponent.genserver_name_suffix(dc)
+
+    "headcounts--user_#{assigns.user_id}#{suffix}"
```

Comment:
Same question here about the suffix. I don't think it is needed and if it isn't, please revert this change.